restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/AccountsEndpoint.java

```diff
@@ -376,57 +378,73 @@ protected Response putAccount(final MultivaluedMap<String, String> data, final M
 
         // what if effectiveAccount is null ?? - no need to check since we checkAuthenticatedAccount() in AccountsEndoint.init()
         final Sid sid = userIdentityContext.getEffectiveAccount().getSid();
-        final Account parent = accountsDao.getAccount(sid);
-        Account account = null;
-        try {
-            account = createFrom(sid, data, parent);
-        } catch (IllegalArgumentException  illegalArgumentException) {
-            return status(BAD_REQUEST).entity(illegalArgumentException.getMessage()).build();
-        }catch (final NullPointerException exception) {
-            return status(BAD_REQUEST).entity(exception.getMessage()).build();
-        } catch (PasswordTooWeak passwordTooWeak) {
-            return status(BAD_REQUEST).entity(buildErrorResponseBody("Password too weak",responseType)).type(responseType).build();
-        }
 
-        // If Account already exists don't add it again
+        ExtensionController ec = ExtensionController.getInstance();
+        ApiRequest apiRequest = new ApiRequest(sid.toString(), data, ApiRequest.Type.CREATE_SUBACCOUNT);
+
+        if (executePostApiAction(apiRequest)) {
```

Comment:
is this really post,or should be pre?

Reply:
@jaimecasero good catch, this must be `pre`. I will fix that